### PR TITLE
tests: fix several unstable cases

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

What's Changed:

Make following cases stable

- test_leader_demoted_when_hibernated
- test_merge_isolated_store_with_no_target_peer

And adjust retry policy to avoid request timeout when epoch change.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: tests/integrations/storage/test_storage.rs [hunk 1]
```diff
@@ -673,14 +674,17 @@ pub fn test_txn_store_gc_multiple_keys_cluster_storage(n: usize, prefix: String)
     let (mut cluster, mut store) =
         AssertionStorage::new_raft_storage_with_store_count(3, prefix.as_str());
     let keys: Vec<String> = (0..n).map(|i| format!("{}{}", prefix, i)).collect();
-    for k in &keys {
-        store.put_ok_for_cluster(&mut cluster, k.as_bytes(), b"v1", 5, 10);
-        store.put_ok_for_cluster(&mut cluster, k.as_bytes(), b"v2", 15, 20);
-    }
+    store.batch_put_ok_for_cluster(&mut cluster, &keys, repeat(b"v1" as &[u8]), 5, 10);
```
Comment (BusyJay) [inline tests/integrations/storage/test_storage.rs:677]: /cc @MyonKeminta

File: components/test_backup/src/lib.rs [hunk 2]
```diff
@@ -202,19 +202,32 @@ impl TestSuite {
     }
 
     pub fn must_kv_put(&mut self, key_count: usize, versions: usize) {
```
Comment (overvenus) [inline components/test_backup/src/lib.rs:204]: LGTM

File: tests/integrations/storage/test_storage.rs (hunk 1 above)
Comment (MyonKeminta) [reply tests/integrations/storage/test_storage.rs]: This part LGTM.

**Review by hicqu — APPROVED** [review]

**Comment by hicqu** [comment]
LGTM.

File: components/test_raftstore/src/util.rs [hunk 3]
```diff
@@ -772,29 +773,27 @@ pub fn put_cf_till_size<T: Simulator>(
...
+            len += key.len() as u64 + 1;
+            len += value.len() as u64;
+            reqs.push(new_put_cf_cmd(cf, key.as_bytes(), &value));
         }
+        cluster.batch_put(key.as_bytes(), reqs).unwrap();
+        // Flush memtable to SST periodically, to make approximate size more accurate.
+        cluster.must_flush_cf(cf, true);
     }
-    // Approximate size of memtable is inaccurate for small data,
-    // we flush it to SST so we can use the size properties instead.
```
Comment (NingLin-P) [inline components/test_raftstore/src/util.rs:795]: This comment should not delete

File: tests/integrations/raftstore/test_merge.rs [hunk 4]
```diff
@@ -938,7 +938,10 @@ fn test_merge_isolated_store_with_no_target_peer() {
     let right_on_store3 = find_peer(&right, 3).unwrap().to_owned();
     pd_client.must_remove_peer(right.get_id(), right_on_store3);
 
+    // Ensure snapshot is sent and applied.
+    must_get_equal(&cluster.get_engine(4), b"k4", b"v1");
```
Comment (NingLin-P) [inline tests/integrations/raftstore/test_merge.rs:942]: `must_get_equal(&cluster.get_engine(4), b"k22", b"v22");` will also ensure that

**Review by NingLin-P — COMMENTED** [review]
Rest LGTM

File: tests/integrations/raftstore/test_merge.rs (hunk 4 above)
Comment (BusyJay) [reply tests/integrations/raftstore/test_merge.rs:942]: They are different. `k22` can only ensure snapshot is applied. Additional check ensure snapshot is applied before proposing `put(k22, v22)`, so follower's `matched` must be greater than 0.

File: components/test_raftstore/src/util.rs (hunk 3 above)
Comment (BusyJay) [reply components/test_raftstore/src/util.rs:795]: It's moved to L793.

Comment (NingLin-P) [reply components/test_raftstore/src/util.rs:795]: No, the reason why flush memtable can make approximate size more accurate is not included.

**Review by NingLin-P — APPROVED** [review]
LGTM

**Pull request merged**